chaosazure/__init__.py

```diff
@@ -136,12 +146,16 @@ def init_resource_graph_client(secrets: Secrets) -> ResourceGraphClient:
 # Private functions
 ###############################################################################
 def __get_credentials(creds: dict) -> ServicePrincipalCredentials:
-    credentials = ServicePrincipalCredentials(
-        client_id=creds['azure_client_id'],
-        secret=creds['azure_client_secret'],
-        tenant=creds['azure_tenant_id'],
-        cloud_environment=__get_cloud_env_by_name(creds['azure_cloud'])
-    )
+    if creds['azure_client_secret'] is not None:
+        credentials = ServicePrincipalCredentials(
+            client_id=creds['azure_client_id'],
+            secret=creds['azure_client_secret'],
+            tenant=creds['azure_tenant_id'],
+            cloud_environment=__get_cloud_env_by_name(creds['azure_cloud'])
+        )
+    else:
+        token = dict(accessToken=creds['access_token'])
```

Comment:
Might be worth raising an exception when the token is missing as well as the secret?

Could be the `chaoslib.exceptions.InterruptExecution` which would terminate the execution right away? As we do on AWS.

Reply:
will be discussed as part of https://github.com/chaostoolkit-incubator/chaostoolkit-azure/issues/84